nuxt.config.ts

```diff
@@ -15,4 +20,104 @@ export default defineNuxtConfig({
   colorMode: {
     classSuffix: '',
   },
+  nitro: {
+    esbuild: {
+      options: {
+        target: 'esnext',
+      },
+    },
+    prerender: {
+      crawlLinks: false,
+      routes: ['/'],
+      ignore: ['/hi'],
+    },
+  },
+  app: {
+    head: {
+      viewport: 'width=device-width,initial-scale=1',
+      link: [
+        { rel: 'icon', href: '/favicon.ico', sizes: 'any' },
+        { rel: 'icon', type: 'image/svg+xml', href: '/nuxt.svg' },
+        { rel: 'apple-touch-icon', href: '/apple-touch-icon.png' },
+      ],
+      meta: [
+        { name: 'viewport', content: 'width=device-width, initial-scale=1' },
+        { name: 'description', content: name },
+        { name: 'apple-mobile-web-app-status-bar-style', content: 'black-translucent' },
+      ],
+    },
+  },
+  pwa: {
+    registerType: 'autoUpdate',
+    scope: '/',
+    base: '/',
+    manifest: {
+      id: '/',
+      scope: '/',
+      name,
+      short_name: name,
+      description: name,
+      theme_color: '#ffffff',
+      icons: [
+        {
+          src: 'pwa-192x192.png',
+          sizes: '192x192',
+          type: 'image/png',
+        },
+        {
+          src: 'pwa-512x512.png',
+          sizes: '512x512',
+          type: 'image/png',
+        },
+        {
+          src: 'maskable-icon.png',
+          sizes: '512x512',
+          type: 'image/png',
+          purpose: 'any maskable',
+        },
+      ],
+    },
+    workbox: {
+      globPatterns: ['**/*.{js,css,html,txt,png,ico,svg}'],
+      navigateFallbackDenylist: [/^\/api\//],
+      navigateFallback: '/',
+      cleanupOutdatedCaches: true,
```

Comment:
These could also go defaults

Reply:
I'll create `config` folder and add pwa and constants modules: I think the pwa module should be reviewed, no makes sense to add some defaults, for example, if the user decides to use `injectManifest` strategy, `workbox` entry will not be required.